src/os/ogc/query/ogcquery.js

```diff
@@ -0,0 +1,143 @@
+goog.module('os.ogc.query.OGCQuery');
+
+goog.require('os.ui.ogc.OGCListUI'); // touch the <ogclist> directive to get it to inject...
+goog.require('plugin.areadata'); // touch the AreaData plugin constants to bring the code into core OS
+
+const query = goog.require('os.query');
+const ConfirmUI = goog.require('os.ui.window.ConfirmUI');
+
+const Feature = goog.requireType('ol.Feature');
+const OGCService = goog.requireType('os.ogc.OGCService');
+
+
+/**
+ * Generic UI and handlers for an OGC Query
+ */
+class OGCQuery {
+  /**
+   * @param {!OGCService} service
+   */
+  constructor(service) {
+    /**
+     * @type {OGCService}
+     */
+    this.service = service;
+
+    /**
+     * @type {?pluginx.areadata.ListUIOptions}
+     * @protected
+     */
+    this.options_;
+
+    /**
+     * @type {string}
+     * @protected
+     */
+    this.eventType_ = 'ogcQuery';
+  }
+
+  /**
+   * Handle a area being chosen in the import area border dialog.
+   * @param {function(!Feature, !Feature)} callback The callback to fire when the area is loaded
+   * @param {Array<!Feature>} data The requested area (array because select2 supports multi-select)
+   * @private
+   */
+  onChoice_(callback, data) {
+    if (data && data.length > 0) {
+      if (this.service) {
+        // inject the requested area into the callback so it has some additional context
+        this.service
+            .get(data[0])
+            .then(callback.bind(undefined, data[0]));
+      }
+    }
+  }
+
+  /**
+   * Handle area being loaded from the server. Insert into the Area manager.
+   * @param {!Feature} datum The requested area
+   * @param {!Feature} feature The area border feature
```

Comment:
Is the distinction between these two arguments still necessary? Seems confusing if we're just creating a feature.

Reply:
Removed!